### ISSUE TITLE
Select Camping Location

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-public#42]
#### Description:

[User Story]

select sub-area (campground)

[Description of the ticket]

#### Acceptance Criteria:

Given
When
Then 

Given
When
Then

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by
- Blocking

**Wireframe(s):**

**Workflows:**

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
Full refinement depends on the wireframe being completed.

**Comment by danieltruong** [comment]
UIs have been stubbed: 

![Image](https://github.com/user-attachments/assets/c6890c30-b3aa-4619-8a43-438bb9a0ec8e)

Moving to blocked. At the moment there is no data being pulled. Currently waiting on the resolution of https://github.com/bcgov/reserve-rec-api/issues/69.

**Comment by danieltruong** [comment]
According to the current wireframes for first pass, there is no way to get to this page. Consider the current state of the search result (2025-04-17):

![Image](https://github.com/user-attachments/assets/881673f2-696e-44ee-9140-ec9f596dcbf4)

In the image, the search result shows "Park names". This implies that when searching, we can only get results related to Park objects. The closest resemblance to a "park" in our data model is Protected Areas found here https://github.com/bcgov/reserve-rec-api/wiki/Data-Model#protected-areas. Our backend and Opensearch is able to support this, however, we have nothing in our wireframe to display details of a park.

Consider the following screenshot as the page that needs to be implemented by this ticket:

![Image](https://github.com/user-attachments/assets/f4b09bd2-a7e8-4cdd-928a-4585308c43ac)

In the screenshot, you can see that facility name is the subtitle of this page. The inclusion of this subtitle implies that this page will at least need the facility object (https://github.com/bcgov/reserve-rec-api/wiki/Data-Model#facility-properties). However, this does not work with the previous search page only allowing parks to appear as a result. Within a protected area, you can have multiple facilities. Therefore, clicking on a Park in the search result cannot go to the page being built in this ticket, because, we do not know which facility details to display.

I have two recommendations:

- Create a park details page that displays all its facilities. Link the current search results to said park details pages. Have the facilities found on the park details page to link to the facility details page (what is being built in this ticket)
- Set the results of Opensearch to be filtered to display only facility type objects that directly link to facility details pages. 

For MVP, I would recommend a state where our search display PAs, facilities, geozones and activities and they all go to their respective details pages as there is an inherent hierarchy in which our data is organized. https://app.mural.co/t/bcparks2575/m/dds5274/1731535677616/06b2ecf8407b6736c15629da1a05a057b5f49f86
At the moment, our Opensearch is able to display all these datatypes based off of your text search.

@Dianadec @marklise @LindsayMacfarlane @pavelcreates @jessicamaclean

**Comment by danieltruong** [comment]
Another flag referencing the current wireframes as of 2025-04-17:

![Image](https://github.com/user-attachments/assets/26935c23-e29e-4082-85f2-ebd259aa65bb)

At the moment, this page implies there is a one to one mapping between facilities (https://github.com/bcgov/reserve-rec-api/wiki/Data-Model#facilities) and activities (https://github.com/bcgov/reserve-rec-api/wiki/Data-Model#activities-1). That will soon not be the case once we add any other data type.

That being said, as far as I know, Backcountry Registration facilities only ever have one single activity associated with them. Let me know if that is actually not the case @jessicamaclean @cameronpettit.

In the current design, I will assume that is the case, however, we will want to look into revising our designs to accommodate multiple possible activities.

@Dianadec @marklise @LindsayMacfarlane @pavelcreates

**Comment by cameronpettit** [comment]
@danieltruong to answer your question on the number of activities per facility: If we were building a system that only handles backcountry registrations and will never handle any other type of activity, then its true that any facility will have at most one activity associated with it (because the system itself only has one activity associated with it). 

However, as per the data model, facilities have a none-to-many relationship with activities, and building the system out under the assertion that there will only ever be one activity (to simplify development for backcountry registration) will bar us from scaling with ease when we run into cases where more than one activity applies to the same facility. If we are building the system in the context that it will one day inherit more offerings than just backcountry registration, I would rather build the system in a way that is able to handle this none-to-many relationship right off the bat.

The example I always like to use is Cheakamus Lake campground. It is a backcountry registration campground but it also can be accessed as part of the day use offering. To answer the user need of "I want to visit Cheakamus campground", we need to be able to distinguish what offerings such a facility has. Do they want to visit in a backcountry registration context or in a day use context? They will need a way to choose this when looking at a facility details page. 

In my opinion, for minimal additional effort we could refactor this page to be interpreted as "Here's all the activities offered at this facility" and only list the one for backcountry registration for MVP, instead of "here is the single activity offered at this facility", and then be forced to refactor to the multiple-activity case later down the line. @pavelcreates @LindsayMacfarlane @marklise

**Comment by LindsayMacfarlane** [comment]
I support the approach that @cameronpettit has indicated: "In my opinion, for minimal additional effort we could refactor this page to be interpreted as "Here's all the activities offered at this facility" and only list the one for backcountry registration for MVP". If there are additional design considerations to make this possible, please reach out to @pavelcreates and myself to walk us through the thoughts/considerations.

**Comment by Dianadec** [comment]
During standup we discussed this and how to get to the facilities page and that a working meeting would be needed as a team. So todo = schedule working meeting to go through the flow and using data points.

**Comment by Dianadec** [comment]
We used today's refinement to clarify the work around this ticket.
Updates:
- adjusted title of this ticket, away from facilities details since this is about the activity level, backcountry registration requirements
- description of this ticket has been updated according to the flow of screen we established were needed to account for the park>facility>activity data model pieces.

Todo: clarify whether this ticket is unblocked, or if it is still blocked on data/dynamodb things.

**Comment by cameronpettit** [comment]
Protected areas, geozones, facilities and activities are available in the database so likely this is partially unblocked. Products will be added soon which should allow us to get to the next booking information pages.

**Comment by LindsayMacfarlane** [comment]
@cameronpettit - thank you for the update. I'll pull this ticket back into 'todo', with the understanding that only certain pieces may be unblocked.

**Comment by cameronpettit** [comment]
I have done my best to adhere to the design spec. There were one or two minor deviations:

1.  The `displayName` of the activity is typically descriptive enough. The design calls for park name, facility name, and activity name to be laid out like this:
![Image](https://github.com/user-attachments/assets/2a7d6066-57fe-4032-bdbd-0f47aaf24959)
However some activities have multiple facilities, so this display won't always be accurate (consider Cape Scott Backcountry Camping where there are multiple campgrounds in the backcountry that all have the same activity associated with them). Additionally to get park and facility information would require two additional backend queries. Not the end of the world but it is an expensive tradeoff considering the benefits.

2. Slight differences in form fields from what is designed as the NGDS-Forms toolkit is used to handle the fields (particularly the complicated one like the date range picker). The styles in the toolkit can not yet be updated. 

3. Some creative liberties were taken with the map, like displaying the geozone boundary related to the activity. This was not a design requirement but it captures the effective area of the activity nicely and helps the map focus on the correct area.

Clicking 'Add booking to cart' does nothing yet, will be handled in #65. It does `console.log()` the booking request. See the PR for more information.

**Comment by manuji** [comment]
Blocked. Search page does not load any parks.

**Comment by manuji** [comment]
Tested on TEST: **Pending**
* Verified that the page has been implemented. 

**Issues:**
1. The input field to add party size accepts decimal values and it is possible to proceed to next screen with decimal values. 
3. The date range accepts past dates to be added and the "Add booking to card" button allows users to proceed to the next screen. 
4. Upon reloading the page, the park name at the top get cleared out. 
5. The option to increase/decrease the input field values doesn't work consistently

Refer to the screen recording below to see the issues mentioned above.

**Comment by cameronpettit** [comment]
> The input field to add party size accepts decimal values and it is possible to proceed to next screen with decimal values.

Changed the party size component to only allow integers 0-10

> The date range accepts past dates to be added and the "Add booking to card" button allows users to proceed to the next screen.

Change the allowable startDate to be equal to or greater than today, whatever today is

> Upon reloading the page, the park name at the top get cleared out.

Can't replicate, please provide more replication instructions

> The option to increase/decrease the input field values doesn't work consistently

Changed the increment/decrement buttons to allow decrement even if the total occupant count is greater than 10

**Comment by manuji** [comment]
@cameronpettit here are the steps to replicate the issue 3. 

1. Navigate to https://d1i5ib4fvf3hku.cloudfront.net/
2. Find a park (ex: Strathcona)
3. Select the Activity "Buttle Lake Marine Sites Backcountry Camping" from the search results. 
4. Refresh the page from the browser refresh option.

**Comment by cameronpettit** [comment]
@manuji some clarification of behaviours for 

> Upon reloading the page, the park name at the top get cleared out.

This is in reference to the search bar at the top of the activities details page, not the activity title itself:

![Image](https://github.com/user-attachments/assets/f9bea0e8-333f-4177-82de-b41dd8df98f2)

Upon refreshing the page, the url route is reloaded and it is treated the same way as if you navigated to that page from anywhere else, so not necessarily from the search on the previous pages. As such the site is unaware of your search terms (theyve been cleared) since the browser assumes you could have gotten to that page from anywhere. This is why this box is empty on a page refresh.

I could add in the search terms to the query parameters so that box could be populated even on a page refresh, but as it is a non-functional search block the overhead work wouldn't really provide a ton of value. If it is important that the value is there, it should probably be addressed in a new ticket.

Additional note - user must be logged in to recreate this issue.
